### DISCUSSION TITLE
Dynamic management of resources allocated to hpx app

### DISCUSSION THREAD

**Discussion by venartas** [STEllAR-GROUP/hpx#6019] [Q&A]
Hello, 

is there a way to dynamically manage resources allocated to hpx application, for example change (reduce/append) number of threads/nodes without restarting with different arguments? Would like to use hpx in the cloud/servers context.

Thanks

**Comment by hkaiser** [comment]
@venartas HPX supports localities connecting and disconnecting late (however you can disconnect only those localities that have connected late to begin with). This also only works for the TCP parcel-port. 

This functionality is demonstrated by the [heartbeat](https://github.com/STEllAR-GROUP/hpx/tree/master/examples/heartbeat) example and the [launch_process](https://github.com/STEllAR-GROUP/hpx/blob/master/libs/full/runtime_components/tests/unit/components/launch_process.cpp)/[launched_process](https://github.com/STEllAR-GROUP/hpx/blob/master/libs/full/runtime_components/tests/unit/components/launched_process.cpp) test.

Please let us know if you need more information.
